heat/core/communicator.py

```diff
@@ -1,25 +1,91 @@
-import torch.distributed as dist
+import abc
+import torch.distributed as mpi
 
+from .stride_tricks import sanitize_axis
 
-def init_dist():
-    dist.init_process_group('mpi')
 
+def initialize_mpi():
+    mpi.init_process_group('mpi')
 
-init_dist()
 
+# initialize the MPI stack
+initialize_mpi()
 
-class NoneCommunicator:
+
+__all__ = [
+    'mpi'
+]
+
+
+class Communicator(metaclass=abc.ABCMeta):
+    @staticmethod
+    @abc.abstractstaticmethod
```

Comment:
`abc.abstractstaticmethod` has become deprecated in Python 3.3, according to the [documentation ](https://docs.python.org/3/library/abc.html#abc.abstractstaticmethod). `abc.abstractmethod` should be sufficient. Btw. is there some special python 3 version we are targeting? Pytorch seems to need at least 3.5 if one chooses python3.